### PR TITLE
QA tool for custom data - DB changes only

### PR DESCRIPTION
**Pull request by mozilloid**
# References:

Jira: Mntor-3423
Figma:

# Description
This PR has only the DB changes from [this](https://github.com/mozilla/blurts-server/pull/4857) pr

# Screenshot (if applicable)

Not applicable.

# How to test

# Checklist (Definition of Done)

- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/db/migrations/20240710214906_qa_custom_brokers.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      table.jsonb("emails").notNullable();
+      table.jsonb("phones").notNullable();
+      table.jsonb("addresses").notNullable();
+      table.jsonb("relatives").notNullable();
+      table.string("first_name").notNullable();
+      table.string("middle_name").nullable();
+      table.string("last_name").notNullable();
+      table.string("status").notNullable();
+      table.boolean("manually_resolved").defaultTo(false);
+      table.timestamp("created_at").defaultTo(knex.fn.now());
```
Comment (mansaj) [inline src/db/migrations/20240710214906_qa_custom_brokers.js:26]: we have another field for "optout_attempts", is that omitted intentionally?

File: src/db/tables/onerep_scans.ts [hunk 2]
```diff
@@ -139,25 +140,46 @@ async function getLatestOnerepScan(
   return scan ?? null;
 }
 
+/*
```
Comment (mansaj) [inline src/db/tables/onerep_scans.ts:143]: @rhelmer I'm a bit wary about merging this code in. 1) We don't know how long this code will be relevant 2) especially when it comes to creating new tables, having data in those tables, I'm not sure what the Postgres restrictions we have, and whether it is easy to delete an entire table if necessary

Comment (mozilloid) [reply src/db/tables/onerep_scans.ts:143]: @mansaj I think that these changes could live as long as onerep is used in any way. Once onerep is fully obsolete, then this code would need to be rewritten for the new provider (which @codemist asked that I write a doc about before the end of my internship). For the db changes, I think this table could live, but some column names would need to be changed depending on how the data object looks

File: src/db/migrations/20240710214906_qa_custom_brokers.js (hunk 1 above)
Comment (mozilloid) [reply src/db/migrations/20240710214906_qa_custom_brokers.js:26]: For what I was trying to achieve, it didn't matter, but I think it's better to add it as well for completeness. Thanks!

**Review by mansaj — APPROVED** [review]

**Pull request merged**